dolphin/api/v1/storages.py

```diff
@@ -132,8 +132,21 @@ def create(self, req, body):
     def update(self, req, id, body):
         return dict(name="Storage 4")
 
+    @wsgi.response(202)
     def delete(self, req, id):
-        return webob.Response(status_int=http_client.ACCEPTED)
+        ctxt = req.environ['dolphin.context']
+        try:
+            device = db.storage_get(ctxt, id)
+        except Exception as e:
+            LOG.error(e)
+            raise exc.HTTPBadRequest(explanation=e.msg)
+        else:
```

Comment:
Need to call DriverManger API remove_storage to remove driver instance for this storage

Reply:
Added it in task.py